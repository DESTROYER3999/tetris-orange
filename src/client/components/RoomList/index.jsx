import { connect } from "react-redux";
import RoomListSub from "./subcomponent";

const mapStateToProps = state => ({
  rooms: state.game.rooms,
  playerName: state.player.name
});

const RoomList = connect(
  mapStateToProps,
  null
)(RoomListSub);

export default RoomList;
