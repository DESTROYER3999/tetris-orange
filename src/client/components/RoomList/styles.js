import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
`;

export const StyledList = styled.div`
  height: 100%;
  border: 3px solid white;
  border-radius: 0px 0px 5px 40px;
  overflow-y: auto;
`;

export const StyledListItem = styled.div`
  height: 5%;
  margin-bottom: 1px;
  margin-top: 1px;
  flex-grow: 1;
  display: flex;
  background-color: rgb(255, 255, 255, 0.1);
  align-items: center;
  transition: background-color 0.3s, font-size 0.3s;
  &:hover {
    background-color: rgb(255, 255, 255, 0.2);
  }
  &:hover #content {
    font-size: 1.8vh;
  }
`;

export const StyledListItemContent = styled.div`
  & #content {
    font-size: 1.5vh;
  }
  display: flex;
  justify-content: center;
  ${css`
    flex: ${props => props.flex};
  `};
`;

export const StyledListSubHeader = styled.div`
  & #header {
    font-size: 2vh;
  }
  height: 6%;
  display: flex;
  align-items: center;
  border: 3px solid white;
  border-bottom: 0px;
  border-radius: 5px 5px 0px 0px;
`;

export const StyledListSubHeaderColumn = styled.div`
  display: flex;
  justify-content: center;
  ${css`
    flex: ${props => props.flex};
  `};
`;
