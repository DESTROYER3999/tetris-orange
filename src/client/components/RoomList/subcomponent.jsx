import React from "react";
import PropTypes from "prop-types";
import {
  StyledContainer,
  StyledList,
  StyledListItem,
  StyledListItemContent,
  StyledListSubHeader,
  StyledListSubHeaderColumn
} from "./styles";
import {
  MAX_PLAYER_BATTLEROYAL,
  MAX_PLAYER_SOLO,
  SOLO
} from "../../../constants/game";

const RoomListSub = ({ rooms, playerName }) => {
  const handleClick = roomName => {
    window.location.hash = "";
    setTimeout(() => {
      window.location.hash = roomName + "[" + playerName + "]";
    }, 20);
  };
  return (
    <StyledContainer id="room-list">
      <h2>Rooms :</h2>

      <StyledListSubHeader id="subheader">
        <StyledListSubHeaderColumn flex={3}>
          <p id="header">Name</p>
        </StyledListSubHeaderColumn>
        <StyledListSubHeaderColumn flex={1}>
          <p id="header">Mode</p>
        </StyledListSubHeaderColumn>
        <StyledListSubHeaderColumn flex={1}>
          <p id="header">Players</p>
        </StyledListSubHeaderColumn>
        <StyledListSubHeaderColumn flex={1}>
          <p id="header">Status</p>
        </StyledListSubHeaderColumn>
      </StyledListSubHeader>
      <StyledList id="list">
        {rooms.map((room, index) => {
          const maxPlayer =
            room.mode === SOLO ? MAX_PLAYER_SOLO : MAX_PLAYER_BATTLEROYAL;
          return (
            <StyledListItem
              id="list-item"
              key={index}
              onClick={() => handleClick(room.name)}
            >
              <StyledListItemContent id="content" flex={3}>
                {room.name}
              </StyledListItemContent>
              <StyledListItemContent id="content" flex={1}>
                {room.mode}
              </StyledListItemContent>
              <StyledListItemContent id="content" flex={1}>
                {room.playersCount}/{maxPlayer}
              </StyledListItemContent>
              <StyledListItemContent id="content" flex={1}>
                {room.isStarted ? "In game" : "In Menu"}
              </StyledListItemContent>
            </StyledListItem>
          );
        })}
      </StyledList>
    </StyledContainer>
  );
};

RoomListSub.propTypes = {
  rooms: PropTypes.array.isRequired
};

export default RoomListSub;
