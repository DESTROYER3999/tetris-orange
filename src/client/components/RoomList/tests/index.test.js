import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import RoomList from "../index";

const mockStore = configureStore();
const initialState = {
  game: {
    rooms: [
      { roomName: "Room1", playersCount: 2 },
      { roomName: "Room2", playersCount: 3 }
    ],
    highscores: []
  },
  player: { name: "Player1" }
};
const store = mockStore(initialState);

describe("<RoomList />", () => {
  test("RoomList rendering with store without crashing", () => {
    const wrapper = shallow(<RoomList store={store} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
