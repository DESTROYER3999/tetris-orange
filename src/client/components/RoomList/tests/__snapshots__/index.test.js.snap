// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomList /> RoomList rendering with store without crashing 1`] = `
<ContextProvider
  value={null}
>
  <RoomListSub
    dispatch={[Function]}
    playerName="Player1"
    rooms={
      Array [
        Object {
          "playersCount": 2,
          "roomName": "Room1",
        },
        Object {
          "playersCount": 3,
          "roomName": "Room2",
        },
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
